iothub/device/src/Transport/Amqp/AmqpConnectionHolder.cs

```diff
@@ -36,7 +37,7 @@ public AmqpConnectionHolder(DeviceIdentity deviceIdentity)
         public AmqpUnit CreateAmqpUnit(
             DeviceIdentity deviceIdentity,
             Func<MethodRequestInternal, Task> onMethodCallback,
-            Action<Twin, string, TwinCollection, IotHubException> twinMessageListener,
+            Action<AmqpMessage, string, IotHubException> twinMessageListener,
```

Comment:
The twin listener is now updated to respond to an amqp message directly. This removes the need to specify a range of params for the different twin response events, i.e. Twin/ClientProperties for the GET operation, TwinCollection/ClientPropertyCollection for responding to property update notifications and ClientPropertyUpdateResponse for receiving the response for a PATCH operation.